ouroboros-consensus-test/src/Test/Util/Serialisation/Roundtrip.hs

```diff
@@ -231,14 +231,29 @@ instance ( blockVersion ~ BlockNodeToClientVersion blk
       Query.TopLevelQueryDisabled ->
         arbitraryBlockQuery queryVersion
 
-      Query.QueryVersion1 ->
+      Query.QueryVersion1 -> genTopLevelQuery queryVersion
+      Query.QueryVersion2 -> genTopLevelQuery queryVersion
+    where
+      genTopLevelQuery queryVersion =
```

Comment:
This will change the distribution of what QC generates, won't it? So we'll go from a `15` to `1` to a `15` to `3` block query to "with version" generated data. Is this desired?

Reply:
What principle should be followed for this kind of decision making?

Reply:
Ideally, we'd have coverage testing which would safeguard that we still cover a sufficient percentage of the cases we consider important. I'll ping you on slack to discuss this further.